### PR TITLE
Feature/search message

### PR DESCRIPTION
**Pull request by braulioPad**
Task: "An endpoint to read the message list for the current user (as identified by their
HTTP Basic authentication credentials). Include messages they have sent and
messages sent by users they follow. Support a “search=” parameter that can be
used to further filter messages based on keyword."

### CODE REVIEW

File: src/main/java/com/main/controller/PersonController.java [hunk 1]
```diff
@@ -1,19 +1,26 @@
...
 
 @RestController
 public class PersonController {
-	
-	
-	@GetMapping
-	public List<Message> MessagesByPerson(){
-		return null;
+	@Autowired
+	private PersonService personService;
```
Comment (FelipeGaher) [inline src/main/java/com/main/controller/PersonController.java:16]: Why not autowiring by constructor?

File: src/main/java/com/main/controller/PersonController.java [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
+
+	@RequestMapping(value = { "/search", "/search/{word}" }, method = RequestMethod.GET)
```
Comment (FelipeGaher) [inline src/main/java/com/main/controller/PersonController.java:22]: Filtering/Search is not recommended b using PathVariables, what other option can you use to Filter/Search?

File: src/main/java/com/main/dto/ResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.List;
+
+import lombok.Data;
+
+
+@Data
+public class ResponseDto<T>{
+	
+	
+	private int status;
```
Comment (FelipeGaher) [inline src/main/java/com/main/dto/ResponseDto.java:9]: Format pls

File: src/main/java/com/main/service/impl/PersonServiceImpl.java [hunk 4]
```diff
@@ -2,32 +2,50 @@
...
 import com.main.entity.Message;
 import com.main.entity.Person;
 import com.main.repository.PersonRepository;
 import com.main.service.PersonService;
 
+@Service
 public class PersonServiceImpl implements PersonService {
 
 	@Autowired
 	private PersonRepository personRepository;
```
Comment (FelipeGaher) [inline src/main/java/com/main/service/impl/PersonServiceImpl.java:19]: @Autowired here neither field DI is needed, use constructor DI.

File: src/main/java/com/main/service/impl/PersonServiceImpl.java [hunk 5]
```diff
@@ -2,32 +2,50 @@
...
+		// If the person is found, retrieve all messages from that person
+		if (person != null) {
+			List<Message> result = personRepository.findAllMessagesByPersonIdAndSearchWord(personId, word);
+			if (!result.isEmpty()) {
+				response.setStatus(200);
+				response.setMessage("Data retrieved successfully");
+				response.setData(result.stream().map(Message::getContent).collect(Collectors.toList()));
+				return response;
+			} else {
+				response.setStatus(204);
```
Comment (FelipeGaher) [inline src/main/java/com/main/service/impl/PersonServiceImpl.java]: 200  Empty is OK, 204 is used when using POST/PUT/DELETE to indicate the action was performed

File: src/main/java/com/main/service/impl/PersonServiceImpl.java [hunk 6]
```diff
@@ -2,32 +2,50 @@
...
+				response.setMessage("No Data found");
+				response.setData(Collections.emptyList());
+				return response;
+			}
+		}
+		response.setStatus(204);
```
Comment (FelipeGaher) [inline src/main/java/com/main/service/impl/PersonServiceImpl.java]: 200

**Comment by FelipeGaher** [comment]
Additionally
- The Challenges requires to not use ORM, instead use JdbcTemplate.

File: src/main/java/com/main/controller/PersonController.java (hunk 1 above)
Comment (braulioPad) [reply src/main/java/com/main/controller/PersonController.java:16]: It's the old way in Spring, but I will change to Constructor,

**Pull request closed without merging**